setup.py

```diff
@@ -231,8 +231,14 @@ def run(self):
                       '-fno-strict-aliasing']
 if os.getenv('PYTORCH_BINARY_BUILD') and platform.system() == 'Linux':
     print('PYTORCH_BINARY_BUILD found. Static linking libstdc++ on Linux')
-    extra_compile_args += ['-static-libstdc++']
-    extra_link_args += ['-static-libstdc++']
+    # get path of libstdc++ and link manually.
+    # for reasons unknown, -static-libstdc++ doesn't fully link some symbols
+    CXXNAME = 'g++'
+    if os.getenv('CXX'):
+        CXXNAME = os.getenv('CXX')
+    path = subprocess.check_output([CXXNAME, '-print-file-name=libstdc++.a'])
```

Comment:
Is this going to work in Python3? `check_output` returns a byte-string. The `replace` call below will fail since the arguments are regular (unicode) strings.